src/main/java/WayofTime/bloodmagic/ritual/portal/Teleports.java

```diff
@@ -115,28 +124,70 @@ public void teleport() {
-//                            server.getConfigurationManager().transferPlayerToDimension(player, newWorldID, new TeleporterBloodMagic(newWorldServer));
-                            player.setPositionAndUpdate(x + 0.5, y + 0.5, z + 0.5);
-                            player.getEntityWorld().updateEntityWithOptionalForce(player, false);
-                            player.connection.sendPacket(new SPacketUpdateHealth(player.getHealth(), player.getFoodStats().getFoodLevel(), player.getFoodStats().getSaturationLevel()));
+
+
+                            /* begin brandon3055 "BrandonsCore" intedimensional teleportation code */
+
+                            PlayerList playerList = server.getPlayerList();
+
+                            player.dimension = newWorldID;
+                            player.connection.sendPacket(new SPacketRespawn(player.dimension, newWorldServer.getDifficulty(), newWorldServer.getWorldInfo().getTerrainType(), player.interactionManager.getGameType()));
+                            playerList.updatePermissionLevel(player);
+                            oldWorldServer.removeEntityDangerously(player);
+                            player.isDead = false;
+
+                            //region Transfer to world
+
+                            player.setLocationAndAngles(x + 0.5, y + 0.5, z + 0.5, player.rotationYaw, player.rotationPitch);
+                            player.connection.setPlayerLocation(x + 0.5, y + 0.5, z + 0.5, player.rotationYaw, player.rotationPitch);
+                            newWorldServer.spawnEntity(player);
+                            newWorldServer.updateEntityWithOptionalForce(player, false);
+                            player.setWorld(newWorldServer);
+
+                            //endregion
+
+                            playerList.preparePlayer(player, oldWorldServer);
+                            player.connection.setPlayerLocation(x + 0.5, y + 0.5, z + 0.5, player.rotationYaw, player.rotationPitch);
+                            player.interactionManager.setWorld(newWorldServer);
+                            player.connection.sendPacket(new SPacketPlayerAbilities(player.capabilities));
+
+                            playerList.updateTimeAndWeatherForPlayer(player, newWorldServer);
+                            playerList.syncPlayerInventory(player);
+
+                            for (PotionEffect potioneffect : player.getActivePotionEffects()) {
+                                player.connection.sendPacket(new SPacketEntityEffect(player.getEntityId(), potioneffect));
+                            }
+                            FMLCommonHandler.instance().firePlayerChangedDimensionEvent(player, entity.dimension, newWorldID);
+                            player.setLocationAndAngles(x + 0.5, y + 0.5, z + 0.5, player.rotationYaw, player.rotationPitch);
+
+                            /* end brandon3055 teleportation code */
+
+                            ForgeChunkManager.releaseTicket(chunkTicket);
```

Comment:
The ticket is only being released when a player is being teleported. Any other mob will retain the ticket.